drivers/vfs/driver.go

```diff
@@ -244,6 +248,41 @@ func (d *Driver) Remove(id string) error {
 	return system.EnsureRemoveAll(d.dir(id))
 }
 
+func (d *Driver) GetTempDirRootDirs() []string {
+	tempDirs := []string{filepath.Join(d.home, tempDirName)}
+	// VFS can have layers in different locations through additionalHomes
+	// Add temp directories for each additional home
+	for _, additionalHome := range d.additionalHomes {
+		tempDirs = append(tempDirs, filepath.Join(additionalHome, tempDirName))
+	}
+	return tempDirs
+}
+
+func (d *Driver) DeferredRemove(id string) (tempdir.CleanupTempDirFunc, error) {
+	layerDir := d.dir(id)
+	tempDirRoot := ""
+	tempDirRoots := d.GetTempDirRootDirs()
+	for _, potentialTempDirRoot := range tempDirRoots {
+		homeDir := filepath.Dir(potentialTempDirRoot)
+		if strings.HasPrefix(layerDir, homeDir) {
```

Comment:
`HasPrefix` is _very_ imprecise for paths.

Two possible ways to model this:
- `d.dir2` knows exactly where it found the data, so it can return the “top-level home” it found the layer in.
- This function could hard-code the knowledge that layers exist in `${home}/dir/${id}`.